app/assets/javascripts/backbone/views/editable_table_view.js.coffee

```diff
@@ -0,0 +1,91 @@
+class MS.Views.EditableTableView extends Backbone.View
+
+  el: "table.editable-table"
+
+  editableTableInit: (e) ->
+    $('.editable-table tbody').sortable({
+      handle: "[data-action='move']"
+    })
+
+  addRow: (e) ->
+    e.preventDefault()
+    $button = $(e.currentTarget)
+    $table = $button.closest('table')
+    $new_row = $table.find('tr.hidden').clone()
+    $new_row.removeClass('hidden')
+    $table.append($new_row)
+
+  removeRow: (e) ->
+    e.preventDefault()
+    $row = $(e.currentTarget).closest('tr')
+    $row.remove()
+
+  saveTableData: (e) ->
+    $section = $(e.currentTarget)
+    $tables = $section.find('.editable-table')
+    self = @
+
+    # Set the master input for the table data to empty
+    $section.find('.editable-tables').find('[data-container]').each (index) ->
+      $input = $(this)
+      $input.val("{}")
+
+    # Save the new data to each master input
+    $section.find('.editable-table').each (index) ->
+      $table = $(this)
+      tableKey = $table.data('table')
+      $rows = $table.find('tbody').find('tr')
+      tableData = []
+
+      for key,row of $rows
+        if !isNaN(key)
+          $row = $(row)
+
+          rowResponse = switch tableKey
+            when 'fixed-costs' then self.formatFixedCostsInput($row)
+            when 'products' then self.formatProductsInput($row)
+
+          if rowResponse.rowData
+            tableData.push(rowResponse.rowData)
+
+      # Save generated table data to the master input, whose value is sent to server
+      $masterInput = $table.closest('.editable-tables').find('[data-container]')
+      masterInputValue = JSON.parse($masterInput.val())
+      masterInputValue["#{tableKey}"] = tableData
+      $masterInput.val(JSON.stringify(masterInputValue))
+
+  formatFixedCostsInput: ($row) ->
```

Comment:
I think perhaps this should be called `prepareFixedCostsData`. `Format` suggests formatting stuff for display, to me. Same for the other method below.